source/includes/_endpoint-account.md

```diff
@@ -47,11 +47,114 @@ GET /account/
     "users": "http://app.crunch.io/api/account/users/",
-    "datasets": "http://app.crunch.io/api/account/datasets/"
+    "datasets": "http://app.crunch.io/api/account/datasets/",
+    "applications": "http://app.crunch.io/api/account/applications/"
   }
 }
 ```
 
+##### Applications
+
+```http
+GET /account/applications/
+```
+
+GET returns a Shoji Catalog with the list of all the configured subdomains an account has.
+
+```json
+{
+    "element":"shoji:catalog",
+    "index": {
+        "./mycompany/": {}
+    }
+}
+```
+
+POST a Shoji Entity here to make a new application. The `subdomain` must be
+unique system-wide and must be a valid URL segment; if unavailable or invalid, the server will return a 400 response.
+
+```json
+{
+    "element": "shoji:entity",
+    "body": {
+      "name": "my company",
+      "subdomain": "mycompany",
+      "palette": {
+          "brand": {
+                "system": "#FFAABB", // Color of links, interactable things
+                "data": "#G4EEBB", // Titles and such
+                "warning": "#BAA5E7"
+            }
+      },
+      "manifest": {}
+    }
+}
+```
+
+Attributes `name` and `subdomain` are required; `palette` and `manifest` are optional. Note that you cannot specify logos in the POST request. Use the created entity's `logo/` resource to upload the image files to the
+app (see below).
+
+#### Application entity
+
+```http
+GET /account/applications/1234/
+```
+
+GET this endpoint for a Shoji Entity containing all details about
+the configured application.
+
+```json
+{
+    "element":"shoji:entity",
+    "body": {
+        "name": "Application name",
+        "subdomain": "mycompany",
+        "logos": {
+            "small": "<URL>",
+            "large": "<URL>"
+        },
+        "palette": {
+            "brand": {
+                "system": "#FFAABB", // Color of links, interactable things
+                "data": "#G4EEBB", // Titles and such
+                "warning": "#BAA5E7"
+            }
+        },
+        "manifest": {}
+    },
+    "views": {
+        "logo": "https://app.crunch.io/api/account/applications/mycompany/logo/"
+    }
+}
+```
+
+PATCH this endpoint to change the name, palette, or manifest. Logos are controlled by the logo subresource.
+
+#### Change application logo
+
+```http
+POST /account/applications/1234/logo/
+```
+
+To set/change an application's logo the client needs to make a `multipart/form-data`
+request containing either or both `large` and `small` fields containing the desired
+image files to use. Only account admins are authorized to change this resource.
+
+
+```http
+POST /projects/6c01/icon/ HTTP/1.1
+Content-Disposition: form-data; name="large"; filename="newlogo.jpg"
+Content-Type: image/jpeg
+```
+
+```http
+HTTP/1.1 201 Created
+Location: https://app.crunch.io/api/account/applications/1234/
+```
+
+The server will update the images accordingly and validate
+size/dimensions/file format etc.
```

Comment:
What are the validation criteria/constraints??? What happens if the uploaded file is outside the valid size/dimensions.